launcher/minecraft/Component.h

```diff
@@ -16,13 +19,49 @@ class VersionList;
 }  // namespace Meta
 class VersionFile;
 
+struct UpdateActionChangeVersion {
+    /// version to change to
+    QString targetVersion;
+};
+struct UpdateActionLatestRecommendedCompatable {
+    /// Parent uid
+    QString parentUid;
+    QString parentName;
+    /// Parent version
+    QString version;
+    ///
+};
+struct UpdateActionRemove {};
+struct UpdateActionImportantChanged {
+    QString oldVersion;
+};
+
+using UpdateActionNone = std::monostate;
+
+using UpdateAction = std::variant<UpdateActionNone,
+                                  UpdateActionChangeVersion,
+                                  UpdateActionLatestRecommendedCompatable,
+                                  UpdateActionRemove,
+                                  UpdateActionImportantChanged>;
+
+struct ModloaderMapEntry {
+    ModPlatform::ModLoaderType type;
+    QStringList knownConfictingComponents;
+};
+static const QMap<QString, ModloaderMapEntry> KNOWN_MODLOADERS{
+    { "net.neoforged", { ModPlatform::NeoForge, { "net.minecraftforge", "net.fabricmc.fabric-loader", "net.fabricmc.fabric-loader" } } },
+    { "net.minecraftforge", { ModPlatform::Forge, { "net.neoforged", "net.fabricmc.fabric-loader", "net.fabricmc.fabric-loader" } } },
+    { "net.fabricmc.fabric-loader", { ModPlatform::Fabric, { "net.minecraftforge", "net.neoforged", "org.quiltmc.quilt-loader" } } },
+    { "org.quiltmc.quilt-loader", { ModPlatform::Quilt, { "net.minecraftforge", "net.neoforged", "net.fabricmc.fabric-loader" } } },
+    { "com.mumfrey.liteloader", { ModPlatform::LiteLoader, {} } }
+};
```

Comment:
Isn't the definition of non-trivial static constants recommended to be inside a cpp file and in the header only the declaration?

Reply:
to the best of my knowledge you can not separate the definition and declaration of a `static const`. trying gives me redefinition errors

Reply:
Ok for me it worked:https://github.com/PrismLauncher/PrismLauncher/pull/2069/files#diff-ff43b34189e94500e7774fa80cec64045c4c008b84e5b59b4ee1ad45670a8c39
and https://github.com/PrismLauncher/PrismLauncher/pull/2069/files#diff-27de15ecbae4a7ebc0641e0a4919b91ba982c587b32e177d8ebeab21e03f438cR508-R512
should I revert that?

Reply:
... I'm silly. of course it needs to be a class member. otherwise static means static to the translation unit ie. every inclusion of the header makes a new copy...